openstudiocore/ruby/openstudio/sketchup_plugin/user_scripts/Reports/OSM_Diagnostic_Script.rb

```diff
@@ -451,6 +451,32 @@ def run(runner, user_arguments)
     else
       puts "no orphaned PortList objects were found"
     end
+  
+    # find and remove orphan LifeCycleCost objects
+    puts ""
+    puts "Removing LifeCycleCost objects that are not connected to any model object"
+    #get all LifeCycleCost objects in the model
+    lcc_objects = model.getObjectsByType("OS:LifeCycleCost".to_IddObjectType)
+    #make an array to store the names of the orphan LifeCycleCost objects
+    orphaned_lcc_objects = Array.new
+    #loop through all LifeCycleCost objects, checking for missing Item Name
+    lcc_objects.each do |lcc_object|
+      if lcc_object.isEmpty(4)
```

Comment:
Yeah, that field is required so the API assumes it is filled it, it will crash if you call it and it is not, see the assert here https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/LifeCycleCost.cpp#L110